docs/Coding-Guidelines/SelfContained.md

```diff
@@ -0,0 +1,46 @@
+# Self-Contained
+
+## What is 'Self-Contained'?
+
+Windows App SDK can be deployed via MSIX or self-contained, aka WinAppSDK/MSIX and WinAppSDK/SelfContained.
+
+See the [Self-Contained spec](https://github.com/microsoft/WindowsAppSDK/pull/1202) for more details.
+
+## How to detect if Windows App SDK is deployed Self-Contained?
+
+If necessary, use `Microsoft::WindowsAppSDK::SelfContained::IsSelfContained()` to detect if the
+current process is using WindowsAppSDK deployed via SelfContained (vs MSIX).
+
+**NOTE:** The implementation assumes Windows App SDK is in use. Failure to detect
+WindowsAppSDK/MSIX is thus assumed to be WindowsAppSDK/SelfContained.
+
+```c++
+#include <Microsoft.WindowsAppSDK.SelfContained.h>
+
+void JustDoIt()
+{
+    if (::Microsoft::WindowsAppSDK::SelfContained::IsSelfContained())
+    {
+        printf("WindowsAppSDK/SelfContained");
+    }
+    else
+    {
+        printf("WindowsAppSDK/MSIX");
```

Comment:
"WindowsAppSDK/FrameworkPackage" would be clearer, as you can have a SelfContained MSIX

Reply:
This is specifically referring to how WinAppSDK is deployed - via MSIX or Self-Contained.
How the caller/user/app is deployed is a different question

Phrasing folks have thrown around various, but WinAppSDK/MSIX vs WinAppSDK/SelfContained has been one of the more common and unambiguous ones.

[Initial draft of AppLocal Deployment spec.](https://github.com/microsoft/WindowsAppSDK/pull/1202) (#1202) should define the appropos terminology but that's a draft and a bit dated (still uses AppLocal terminology). So until that's updated I'll stick with the terms here

Reply:
+@mikebattista for visibility